src/hardware/tandy_sound.cpp

```diff
@@ -659,30 +698,28 @@ void TANDYSOUND_Init(Section *section)
 		return;
 	}
 
-	ConfigProfile cfg;
-	switch (machine) {
-	case MCH_PCJR: cfg = ConfigProfile::PCjrSystem; break;
-	case MCH_TANDY: cfg = ConfigProfile::TandySystem; break;
-	default: cfg = ConfigProfile::SoundCardOnly; break;
-	}
-
 	// The second DMA controller conflicts with the tandy sound's base IO
 	// ports 0xc0. Closing the controller itself means that all the high DMA
 	// ports (4 through 7) get automatically shutdown as well.
 	//
 	DMA_ShutdownSecondaryController();
 
+	// Maybe initialize the DAC if requested
+	toggle_dac(section);
+
+	const auto config_profile = get_config_profile();
+
 	const auto wants_dac = has_true(pref) || (IS_TANDY_ARCH && pref == "auto");
-	if (wants_dac) {
-		tandy_dac = std::make_unique<TandyDAC>(
-		        cfg, prop->Get_string("tandy_dac_filter"));
-	}
-	tandy_psg = std::make_unique<TandyPSG>(cfg,
-	                                       wants_dac,
-	                                       prop->Get_string("tandy_filter"));
+
+	tandy_psg_instance = std::make_unique<TandyPSG>(
+	        config_profile, wants_dac, prop->Get_string("tandy_filter"));
 
 	set_tandy_sound_flag_in_bios(true);
+}
 
-	constexpr auto changeable_at_runtime = true;
-	section->AddDestroyFunction(&TANDYSOUND_ShutDown, changeable_at_runtime);
+void TANDYSOUND_ShutDown(Section*)
+{
+	// shutdown with an empty section
+	constexpr Section* empty_section = nullptr;
+	TANDYSOUND_Init(empty_section);
```

Comment:
Ok... so that's how `TANDYSOUND_Init` can receive a null section.

I'm not in love with this as it feels very much like a hack. I'd rather have some code duplication here and make the shutdown process explicit. Surely not everything from init is needed for the shutdown. Or extract the commonallities between init and shutdown into a function, but frankly, that's a bit overdoing it. Some duplication and more explicitness in this case is fine as it would improve readability a lot.